Implement the search index.

mithrandi opened the pull request:
<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/fusionapp/fusion-index/5)
<!-- Reviewable:end -->

jonathanj commented:
Reviewed 3 of 5 files at r1, 2 of 2 files at r2, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

<sup>**[fusion_index/resource.py, line 78 \[r2\]](https://reviewable.io:443/reviews/fusionapp/fusion-index/5#-Jzpqajpo_9WqQGyHqYp)** ([raw file](https://github.com/fusionapp/fusion-index/blob/93afc4b3e71099335a778bd01a280d2c62f13c82/fusion_index/resource.py#L78)):</sup>
I don't think we're handling the case where `searchClass` is not in the enum here.

---

<sup>**[fusion_index/search.py, line 25 \[r2\]](https://reviewable.io:443/reviews/fusionapp/fusion-index/5#-Jzpy9_f5-Qm7QDIxgIy)** ([raw file](https://github.com/fusionapp/fusion-index/blob/93afc4b3e71099335a778bd01a280d2c62f13c82/fusion_index/search.py#L25)):</sup>
This first sentence here looks like it's incomplete.

---

<sup>**[fusion_index/search.py, line 44 \[r2\]](https://reviewable.io:443/reviews/fusionapp/fusion-index/5#-JzpyxrBIIGto5Urtl26)** ([raw file](https://github.com/fusionapp/fusion-index/blob/93afc4b3e71099335a778bd01a280d2c62f13c82/fusion_index/search.py#L44)):</sup>
I noticed when doing the review for the Fusion side of the lookup index that the production name defaults to "prod" not "production" as this example here might indicate. I think we should standardise on our use, even if this is only serving as an example.

---

<sup>**[fusion_index/test/test_resource.py, line 103 \[r2\]](https://reviewable.io:443/reviews/fusionapp/fusion-index/5#-JzqU0ryvz4uzgOV-kzY)** ([raw file](https://github.com/fusionapp/fusion-index/blob/93afc4b3e71099335a778bd01a280d2c62f13c82/fusion_index/test/test_resource.py#L103)):</sup>
I think you should name these functions `GET, `PUT` and `DELETE` since there is existing code in the tests that assigns the names `get` and `put` to log message objects, it's a little confusing to shadow these like this even if the two aren't used in the same test.

---

<sup>**[fusion_index/test/test_resource.py, line 301 \[r2\]](https://reviewable.io:443/reviews/fusionapp/fusion-index/5#-JzqUse2P6_5LPiIMTM8)** ([raw file](https://github.com/fusionapp/fusion-index/blob/93afc4b3e71099335a778bd01a280d2c62f13c82/fusion_index/test/test_resource.py#L301)):</sup>
There don't seem to be any tests for logging prefix searches.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/fusionapp/fusion-index/5)
<!-- Sent from Reviewable.io -->

mithrandi commented:
Review status: 3 of 7 files reviewed at latest revision, 5 unresolved discussions, some commit checks pending.

---

<sup>**[fusion_index/resource.py, line 79 \[r2\]](https://reviewable.io:443/reviews/fusionapp/fusion-index/5#-Jzpqajpo_9WqQGyHqYp)** ([raw file](https://github.com/fusionapp/fusion-index/blob/93afc4b3e71099335a778bd01a280d2c62f13c82/fusion_index/resource.py#L79)):</sup>
I made the code catch the exception and return `NotFound`.

---

<sup>**[fusion_index/search.py, line 25 \[r2\]](https://reviewable.io:443/reviews/fusionapp/fusion-index/5#-Jzpy9_f5-Qm7QDIxgIy)** ([raw file](https://github.com/fusionapp/fusion-index/blob/93afc4b3e71099335a778bd01a280d2c62f13c82/fusion_index/search.py#L25)):</sup>
Added some missing prose here.

---

<sup>**[fusion_index/search.py, line 44 \[r2\]](https://reviewable.io:443/reviews/fusionapp/fusion-index/5#-JzpyxrBIIGto5Urtl26)** ([raw file](https://github.com/fusionapp/fusion-index/blob/93afc4b3e71099335a778bd01a280d2c62f13c82/fusion_index/search.py#L44)):</sup>
Good point, I changed it here as well as in the lookup docs.

---

<sup>**[fusion_index/test/test_resource.py, line 103 \[r2\]](https://reviewable.io:443/reviews/fusionapp/fusion-index/5#-JzqU0ryvz4uzgOV-kzY)** ([raw file](https://github.com/fusionapp/fusion-index/blob/93afc4b3e71099335a778bd01a280d2c62f13c82/fusion_index/test/test_resource.py#L103)):</sup>
I uppercased the names.

---

<sup>**[fusion_index/test/test_resource.py, line 301 \[r2\]](https://reviewable.io:443/reviews/fusionapp/fusion-index/5#-JzqUse2P6_5LPiIMTM8)** ([raw file](https://github.com/fusionapp/fusion-index/blob/93afc4b3e71099335a778bd01a280d2c62f13c82/fusion_index/test/test_resource.py#L301)):</sup>
I added some logging testing to `test_exactAndPrefix` which does prefix tests, so there's a bit more coverage now.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/fusionapp/fusion-index/5)
<!-- Sent from Reviewable.io -->

jonathanj commented:
Reviewed 4 of 4 files at r4.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/fusionapp/fusion-index/5)
<!-- Sent from Reviewable.io -->
